### PR TITLE
Fix the overflow test for fee and block_reward in blocks

### PR DESCRIPTION
**Pull request by Doy-lee**
Fixes `gen_uint_overflow_1` test, has been tested with the reward fudge PR.

Had to change the loki_chain_generator API to work the way this test wanted to use blockchain data, so the bulk of changes is renaming.

@jagerman

### CODE REVIEW

File: tests/core_tests/chaingen.h [hunk 1]
```diff
@@ -752,7 +752,7 @@ struct push_core_event_visitor: public boost::static_visitor<bool>
     log_event("loki_blockchain_addable<cryptonote::checkpoint_t>");
     cryptonote::Blockchain &blockchain = m_c.get_blockchain_storage();
     bool added = blockchain.update_checkpoint(entry.data);
-    CHECK_AND_NO_ASSERT_MES(added == entry.can_be_added_to_blockchain, false, entry.fail_msg);
+    CHECK_AND_NO_ASSERT_MES(added == entry.can_be_added_to_blockchain, false, (entry.fail_msg.size() ? entry.fail_msg : "Failed to add checkpoint (no reason given)"));
```
Comment (jagerman) [inline tests/core_tests/chaingen.h]: These seem unrelated to this PR (they seem fine, just checking that including them was intentional).

Comment (Doy-lee) [reply tests/core_tests/chaingen.h]: Yea
